pkg/operator/staticpod/controller/installer/installer_controller_test.go

```diff
@@ -2084,22 +2085,123 @@ func TestInstallerController_manageInstallationPods(t *testing.T) {
 		want    bool
 		wantErr bool
 	}{
-		// TODO: Add test cases.
+		{
+			name: "if the revision shouldn't install and a new revision is to be rolled out, requeue the new installer pod",
+			fields: fields{
+				targetNamespace: "test-namespace",
+				staticPodName:   "test-pod",
+				command:         []string{},
+				kubeClient:      fake.NewSimpleClientset(),
+				eventRecorder:   eventstesting.NewTestingEventRecorder(t),
+				shouldRevisionInstall: func() (bool, error) {
+					return false, fmt.Errorf("revision shouldn't be installed")
+				},
+			},
+			args: args{
+				operatorSpec: &operatorv1.StaticPodOperatorSpec{
+					OperatorSpec: operatorv1.OperatorSpec{},
+				},
+				originalOperatorStatus: &operatorv1.StaticPodOperatorStatus{
+					LatestAvailableRevision: 1,
+					NodeStatuses: []operatorv1.NodeStatus{
+						{
+							NodeName:        "test-node-1",
+							CurrentRevision: 1,
+							TargetRevision:  2,
+							LastFailedCount: 1,
+						},
+					},
+				},
+			},
+			want:    true,
+			wantErr: true,
+		},
+		{
+			name: "if the revision shouldn't install and no new revision is to be rolled out, then no requeue",
+			fields: fields{
+				targetNamespace: "test-namespace",
+				staticPodName:   "test-pod",
+				command:         []string{},
+				kubeClient:      fake.NewSimpleClientset(),
+				shouldRevisionInstall: func() (bool, error) {
+					return false, fmt.Errorf("revision shouldn't be installed")
+				},
+			},
+			args: args{
+				operatorSpec: &operatorv1.StaticPodOperatorSpec{
+					OperatorSpec: operatorv1.OperatorSpec{},
+				},
+				originalOperatorStatus: &operatorv1.StaticPodOperatorStatus{
+					LatestAvailableRevision: 1,
+					NodeStatuses: []operatorv1.NodeStatus{
+						{
+							NodeName:        "test-node-1",
+							CurrentRevision: 1,
+							TargetRevision:  1,
+						},
+					},
+				},
+			},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name: "if the revision is safe to install and a new revision is to be rolled out, no requeue",
+			fields: fields{
+				targetNamespace: "test-namespace",
+				staticPodName:   "test-pod",
+				configMaps:      []revision.RevisionResource{{Name: "test-config"}},
+				command:         []string{},
+				kubeClient:      fake.NewSimpleClientset(),
+				eventRecorder:   eventstesting.NewTestingEventRecorder(t),
+				installerPodImageFn: func() string {
+					return "docker.io/foo/bar"
+				},
+				shouldRevisionInstall: func() (bool, error) {
+					return true, nil
+				},
+			},
+			args: args{
+				operatorSpec: &operatorv1.StaticPodOperatorSpec{
+					OperatorSpec: operatorv1.OperatorSpec{},
+				},
+				originalOperatorStatus: &operatorv1.StaticPodOperatorStatus{
+					LatestAvailableRevision: 1,
+					NodeStatuses: []operatorv1.NodeStatus{
```

Comment:
I only very superficially looked through your tests, but I'm generally missing one with multiple NodeStatuses. 

Maybe more meaningful for etcd would be a test with three nodes at those revisions:

master-0 = 4
master-1 = 3 (would be next to get to 4)
master-2 = 3

Now our quorum guard says master-0 went away (eg, machine was turned off). What happens to the remaining revisions? 

I would expect this to block the installation of rev 4 on master-1 and master-2. 
Maybe also think of a couple of such tests, maybe with a more structured permutation approach. We can discuss this also next Tuesday.